### ISSUE TITLE
The reproduction results of the CE metric are lower than those reported in the original paper.

### ISSUE THREAD

**Issue by Stella-xixi** [wang-zhanyu/R2GenGPT#18]
Since your code does not include the calculation of the CE metric, I implemented it myself. However, the results I obtained were significantly lower than those reported in the original paper. Could you provide the code for computing the CE metric?
Thanks

**Comment by 20191864218** [comment]
> Since your code does not include the calculation of the CE metric, I implemented it myself. However, the results I obtained were significantly lower than those reported in the original paper. Could you provide the code for computing the CE metric? Thanks

Me too, calculated that the CE metric is much lower, the deep mode of the MIMIC-CXR dataset, the CE metric is "F1_MACOR: 0.267, RECALL_MACRO: 0.254, PRECISION_MACRO: 0.353", what is your CE metric?

**Comment by Abby-simla** [comment]
> > Since your code does not include the calculation of the CE metric, I implemented it myself. However, the results I obtained were significantly lower than those reported in the original paper. Could you provide the code for computing the CE metric? Thanks
> 
> Me too, calculated that the CE metric is much lower, the deep mode of the MIMIC-CXR dataset, the CE metric is "F1_MACOR: 0.267, RECALL_MACRO: 0.254, PRECISION_MACRO: 0.353", what is your CE metric?

https://github.com/wang-zhanyu/R2GenGPT/issues/18#issuecomment-3227317961

**Comment by Stella-xixi** [comment]
> > 由于您的代码没有包含 CE 指标的计算，所以我自己实现了它。但是，我得到的结果明显低于原始论文中报告的结果。您能提供计算 CE 指标的代码吗？谢谢。
> 
> 我也是，计算出 CE 指标低很多，MIMIC-CXR 数据集的深度模式，CE 指标是“F1_MACOR：0.267，RECALL_MACRO：0.254，PRECISION_MACRO：0.353”，你的 CE 指标是什么？

My CE metric is even lower than yours. May I ask which server you ran this result on, and what batch size you set?

**Comment by 20191864218** [comment]
> > > 由于您的代码没有包含 CE 指标的计算，所以我自己实现了它。但是，我得到的结果明显低于原始论文中报告的结果。您能提供计算 CE 指标的代码吗？谢谢。
> > 
> > 
> > 我也是，计算出 CE 指标低很多，MIMIC-CXR 数据集的深度模式，CE 指标是“F1_MACOR：0.267，RECALL_MACRO：0.254，PRECISION_MACRO：0.353”，你的 CE 指标是什么？
> 
> My CE metric is even lower than yours. May I ask which server you ran this result on, and what batch size you set?

train：2 4090D，batch=4；test：1 4090D，batch=4 and --strategy=deepspeed_stage_2

**Comment by Stella-xixi** [comment]
Thank you very much for your reply.
I would like to ask whether your reproduced results of the NLG metrics match those reported in the original paper?
In my own reproduction, the best BLEU-4 score I achieved was only 0.10.
If your results are better, I would greatly appreciate it if you could share how you conducted the experiment.

I look forward to your insights.

Best regards







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wang-zhanyu/R2GenGPT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年9月16日(星期二) 下午3:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [wang-zhanyu/R2GenGPT] The reproduction results of the CE metric are lower than those reported in the original paper. (Issue #18)



20191864218 left a comment (wang-zhanyu/R2GenGPT#18)
    
由于您的代码没有包含 CE 指标的计算，所以我自己实现了它。但是，我得到的结果明显低于原始论文中报告的结果。您能提供计算 CE 指标的代码吗？谢谢。
  
我也是，计算出 CE 指标低很多，MIMIC-CXR 数据集的深度模式，CE 指标是“F1_MACOR：0.267，RECALL_MACRO：0.254，PRECISION_MACRO：0.353”，你的 CE 指标是什么？
  
My CE metric is even lower than yours. May I ask which server you ran this result on, and what batch size you set?
  
train：2 4090D，batch=4；test：1 4090D，batch=4 and --strategy=deepspeed_stage_2

**Comment by 20191864218** [comment]
> Thank you very much for your reply. I would like to ask whether your reproduced results of the NLG metrics match those reported in the original paper? In my own reproduction, the best BLEU-4 score I achieved was only 0.10. If your results are better, I would greatly appreciate it if you could share how you conducted the experiment. I look forward to your insights. Best regards

I tested the deep training mode on the MIMIC-CXR dataset, and the NLP metrics were only slightly lower than those reported in the paper—almost identical. However, on the IU-Xray dataset, the metric results appeared to be quite unstable, fluctuating between higher and lower values.
In the experiment, I only made modifications to the following code and used the dataset mimic_annotation_all.json provided by R2GenGPT.

`        if args.vis_use_lora:
            peft_config_visual = LoraConfig(
                                    r=args.vis_r,
                                    lora_alpha=args.vis_alpha,
                                    target_modules=["query", "value"],
                                    lora_dropout=args.lora_dropout,
                                    bias="none",
                                    modules_to_save=["classifier"],
                                )
            self.visual_encoder = get_peft_model(self.visual_encoder, peft_config_visual)
            self.visual_encoder.print_trainable_parameters()
            print('Loading vision encoder with LoRA -- Done')
        elif args.freeze_vm:
            for name, param in self.visual_encoder.named_parameters():
                param.requires_grad = False
            print(f'Loading Frozen vision encoder:{args.vision_model} -- Done')
        else:
            for p in self.visual_encoder.parameters():
                p.requires_grad = True
            self.visual_encoder.train()
            print(f'Loading Trainable vision encoder:{args.vision_model} -- Done')`

**Comment by Stella-xixi** [comment]
Hello, I also used the dataset file `mimic_annotation_all.json` provided by R2GenGPT. However, other metrics such as NLG in the testing phase fell far short of the levels reported in the original paper. I would like to ask for your advice here: What type of graphics card (GPU) did you use when you conducted your training? Currently, I am only using a single NVIDIA A6000 GPU for training, and I wonder if this might have an impact on the model's performance in testing. In addition, I would like to request a copy of your complete R2GenGPT code. I hope to identify the root cause of the current issues by comparing it with my code. Your kind sharing would be greatly appreciated. Sincerely

**Comment by CKM-30** [comment]
Hello, could you share the MIMIC-CXR dataset? I have been reviewing a lot of X-ray literature recently, and experiments all involve this dataset, but I couldn’t download it from the official source.
